src/lib/mina_base/user_command.ml

```diff
@@ -227,12 +203,44 @@ let valid_until (t : t) =
   | Parties _ ->
       Mina_numbers.Global_slot.max_value
 
-let forget_check (t : Valid.t) : t = (t :> t)
+module Valid = struct
+  [%%versioned
+  module Stable = struct
+    module V2 = struct
+      type t =
+        ( Signed_command.With_valid_signature.Stable.V2.t
+        , Parties.Valid.Stable.V1.t )
+        Poly.Stable.V2.t
+      [@@deriving sexp, compare, equal, hash, yojson]
+
+      let to_latest = Fn.id
+    end
+  end]
+
+  module Gen = Gen_make (Signed_command.With_valid_signature)
+end
+
+let check (t : t) : Valid.t option =
+  match t with
+  | Signed_command x ->
+      Option.map (Signed_command.check x) ~f:(fun c -> Signed_command c)
+  | Parties p ->
+      Some (Parties { Parties.Valid.parties = p; verification_keys = [] })
```

Comment:
This feels like a foot-gun. Shouldn't the default behaviour here be to populate the list?